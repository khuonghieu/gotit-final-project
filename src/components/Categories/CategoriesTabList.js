import { Tab } from '@gotitinc/design-system';
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { chooseCategory, fetchCategories } from '../../actions/categories';
import CurrentCategoryInfo from './CurrentCategoryInfo';

function mapStateToProps(state) {
  return {
    categories: state.categories,
    loggedIn: state.user.loggedIn,
  };
}

const mapDispatchToProps = (dispatch) => ({
  chooseCategory: (category) => dispatch(chooseCategory(category)),
  fetchCategories: (offset, limit) => dispatch(fetchCategories(offset, limit)),
});

export function CategoriesTabList({
  categories, loggedIn, chooseCategory, fetchCategories, history,
}) {
  function onChooseCategory(category) {
    chooseCategory(category);
    history.push(`/catalog/category/${category}`);
  }

  React.useEffect(() => {
    if (loggedIn) {
      fetchCategories(0, 10);
    }
  }, [fetchCategories, loggedIn]);
  return (
    <div>
      <Tab
        current={categories.currentCategory}
        onSelect={onChooseCategory}
      >
        {(categories.categoriesList) ? categories.categoriesList.map((categoryElement) => (<Tab.Item eventKey={categoryElement.id} key={categoryElement.id}>{categoryElement.name}</Tab.Item>)) : 'null'}
      </Tab>
      <CurrentCategoryInfo />
      {/* <CreateCategoryForm /> */}
    </div>
  );
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(CategoriesTabList));
