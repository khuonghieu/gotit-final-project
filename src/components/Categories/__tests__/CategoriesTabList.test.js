import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createBrowserHistory } from 'history';
import { CategoriesTabList } from '../CategoriesTabList';

configure({ adapter: new Adapter() });

describe('components/Categories/CategoriesTabList.js', () => {
  let wrapper;
  let props;
  let tabItems;

  beforeEach(() => {
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
    props = {
      categories: { currentCategory: null, categoriesList: [] },
      loggedIn: false,
      chooseCategory: jest.fn(),
      fetchCategories: jest.fn(),
      history: createBrowserHistory(),
    };
  });

  const update = () => {
    tabItems = wrapper.find('Tab-item');
  };

  const setup = () => {
    wrapper = shallow(
      <CategoriesTabList {...props} />,
    );
    update();
  };
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call fetchCategories when loggedIn is true', () => {
    props = {
      ...props,
      loggedIn: true,
    };
    setup();
    expect(props.fetchCategories).toHaveBeenCalled();
  });
  it('should render correct number of tab items when receiving array of categories', () => {
    props = {
      ...props,
      categories: { currentCategory: null, categoriesList: [{ id: 1, name: 'sport' }, { id: 2, name: 'fashion' }] },
    };
    setup();
    expect(tabItems.length).toBe(2);
  });
});
