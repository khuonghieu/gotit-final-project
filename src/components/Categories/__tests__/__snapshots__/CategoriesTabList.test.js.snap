// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Categories/CategoriesTabList.js should render correctly 1`] = `ShallowWrapper {}`;
